`serialization` and `fdreadoutmodules` don't build in a work area

If I set up a work area based on the nightly, and try to build `serialization` and `fdreadoutmodules` together, I get this error:
```
CMake Error in fdreadoutmodules/CMakeLists.txt:
  Imported target "appfwk::appfwk" includes non-existent path

    "/home/nfs/jcfree/NFD_DEV_250303_A9/build/serialization/codegen/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

--- a/cmake/DAQ.cmake
+++ b/cmake/DAQ.cmake
@@ -347,7 +347,10 @@ function(daq_codegen)
 
   endforeach()
 
-  set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+  if (NOT ${CGOPTS_TEST})
+    set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+  endif()
+
 endfunction()
 
 ####################################################################################################
@@ -522,7 +525,10 @@ function (daq_protobuf_codegen)
   add_custom_target(${PROJECT_NAME}_PROTOBUF_GENERATION DEPENDS ${outfiles}  )
   add_dependencies( ${PRE_BUILD_STAGE_DONE_TRGT} ${PROJECT_NAME}_PROTOBUF_GENERATION)
 
-  set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+  if (NOT ${PROTOBUFOPTS_TEST})
+    set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+  endif()
+
   set(PROTOBUF_FILES ${outfiles} PARENT_SCOPE)
 
 endfunction()
@@ -702,7 +708,11 @@ function(daq_oks_codegen)
    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/oksdalgen_${TARGETNAME} DESTINATION ${CMAKE_INSTALL_DATADIR})
 
   set(DAQ_PROJECT_INSTALLS_TARGETS true PARENT_SCOPE)
-  set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+
+  if (NOT ${config_opts_TEST})
+    set(DAQ_PROJECT_GENERATES_CODE true PARENT_SCOPE)
+  endif()
+
   if(NOT ${config_opts_TEST})
     set(ANY_OKS_FILES ${cpp_source} PARENT_SCOPE)
   else()

